src/librustdoc/clean/mod.rs

```diff
@@ -2735,10 +2735,8 @@ fn clean_maybe_renamed_item<'tcx>(
                 Box::new(clean_ty(ty, cx)),
                 Constant { kind: ConstantKind::Local { body: body_id, def_id } },
             ),
-            ItemKind::OpaqueTy(ref ty) => OpaqueTyItem(OpaqueTy {
-                bounds: ty.bounds.iter().filter_map(|x| clean_generic_bound(x, cx)).collect(),
-                generics: clean_generics(ty.generics, cx),
-            }),
+            // Not sure exactly why this is ok??
```

Comment:
So I think basically this function transforms a hir item into zero or more clean items. If we want to ignore the hir item, we just return an empty list to "delete" it. Unless there was something else you were asking about?

Reply:
> Unless there was something else you were asking about?

Yeah, I'm more wondering why removing the item was fine. But I think notriddles answer above classifies that for me